ui/apps/platform/src/Containers/NetworkGraph/flows/advancedFlowsFilterUtils.ts

```diff
@@ -0,0 +1,42 @@
+import { AdvancedFlowsFilterType, FilterValue } from './types';
+
+function isValidPort(port: string): boolean {
+    const portNum = parseInt(port, 10);
+    if (Number.isNaN(portNum)) {
+        return false;
+    }
+    return portNum >= 1 && portNum <= 65535;
+}
+
+// This function is used to convert our filters data structure to an array of strings
+// which the Select component expects
+export function filtersToSelections(filters: AdvancedFlowsFilterType): FilterValue[] {
+    return Object.values(filters).reduce((acc, curr) => {
+        return [...acc, ...Object.values(curr)];
+    }, []);
+}
+
+// This function is used to convert the selections array of strings, which the Select component
+// expects, into our filters data structure
+export function selectionsToFilters(selections: string[]): AdvancedFlowsFilterType {
+    const filters: AdvancedFlowsFilterType = {
+        flows: [],
+        directionality: [],
+        protocols: [],
+        ports: [],
+    };
+    selections.forEach((selection) => {
+        if (selection === 'anomalous' || selection === 'baseline') {
+            filters.flows.push(selection);
+        } else if (selection === 'ingress' || selection === 'egress') {
+            filters.directionality.push(selection);
+        } else if (selection === 'TCP' || selection === 'UDP') {
+            filters.protocols.push(selection);
+        } else if (isValidPort(selection)) {
+            filters.ports.push(selection);
+        } else {
+            // do nothing
+        }
```

Comment:
why not use a switch statement here and push to filters.ports as default? an empty else clause seems odd to see

Reply:
We can but I'd like to do it like this in case we add something else to the filters. If we don't distinguish each filter, we might accidentally think something is a port that shouldn't be